intranet/apps/printing/views.py

```diff
@@ -349,17 +380,45 @@ def print_job(obj: PrintJob, do_print: bool = True):
         if obj.page_range:
             if not range_count:
                 raise InvalidInputPrintingError("You specified an invalid page range.")
-            elif range_count > settings.PRINTING_PAGES_LIMIT:
+
+            elif range_count > settings.PRINTING_PAGES_LIMIT_TEACHERS and (obj.user.is_teacher or obj.user.is_printing_admin):
                 raise InvalidInputPrintingError(
-                    "You specified a range of {} pages. You may only print up to {} pages using this tool.".format(
-                        range_count, settings.PRINTING_PAGES_LIMIT
-                    )
+                    f"You specified a range of {range_count} pages. "
+                    f"You may only print up to {settings.PRINTING_PAGES_LIMIT_TEACHERS} pages using this tool."
                 )
-        elif num_pages > settings.PRINTING_PAGES_LIMIT:
+
+            elif range_count > settings.PRINTING_PAGES_LIMIT_STUDENTS:
+                raise InvalidInputPrintingError(
+                    f"You specified a range of {range_count} pages. "
+                    f"You may only print up to {settings.PRINTING_PAGES_LIMIT_STUDENTS} pages using this tool."
+                )
+
+        elif num_pages > settings.PRINTING_PAGES_LIMIT_TEACHERS and (obj.user.is_teacher or obj.user.is_printing_admin):
             raise InvalidInputPrintingError(
-                "This file contains {} pages. You may only print up to {} pages using this tool.".format(num_pages, settings.PRINTING_PAGES_LIMIT)
+                f"This file contains {num_pages} pages. " f"You may only print up to {settings.PRINTING_PAGES_LIMIT_TEACHERS} pages using this tool."
             )
 
+        elif num_pages > settings.PRINTING_PAGES_LIMIT_STUDENTS:
+            raise InvalidInputPrintingError(
+                f"This file contains {num_pages} pages. " f"You may only print up to {settings.PRINTING_PAGES_LIMIT_STUDENTS} pages using this tool."
+            )
+
+        if get_user_ratelimit_status(obj.user.username):
+            # Bypass rate limit for admins but still send error message for debugging purposes
+            if obj.user.is_printing_admin:
+                logger.debug(
+                    """Administrator %s passed the rate limit of %s print jobs every %s minutes, but since they are an
+                    administrator the request will still go through.""",
+                    obj.user.username,
+                    settings.PRINT_RATELIMIT_FREQUENCY,
+                    settings.PRINT_RATELIMIT_MINUTES,
+                )
+            # If user needs to be rate limited
+            elif obj.user.is_student:  # Don't rate limit teachers
```

Comment:
This goes back to what we were discussing earlier - **if** printing admin or teacher, bypass, **else** apply ratelimit logic. Not just students.